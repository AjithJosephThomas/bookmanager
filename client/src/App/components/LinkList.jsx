import React from "react";
import { Link } from "react-router-dom";

const LinkList = ({linkCollection,linkPrefix}) => (
  <div className="link-list">
    <ul>
      {linkCollection.map(({ id, name }) => (
        <li key={id}>
          <Link to={`${linkPrefix}${id}`}>{name}</Link>
        </li>
      ))}
    </ul>
  </div>
);
export default LinkList;
