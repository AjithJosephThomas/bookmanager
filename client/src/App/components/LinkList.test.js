import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router'
import LinkList from './LinkList';

describe("LinkList component", () => {
    it('renders LinkList component to render with list', () => {
        const linklist = renderer.create(<MemoryRouter><LinkList linkPrefix={""} linkCollection={[{ id: 1, name: 'test' }]} /></MemoryRouter>).toJSON();
        expect(linklist).toMatchSnapshot();
    });
    it('renders LinkList component to render with empty list', () => {
        const linklist = renderer.create(<MemoryRouter><LinkList linkPrefix={""} linkCollection={[]} /></MemoryRouter>).toJSON();
        expect(linklist).toMatchSnapshot();
    });
})