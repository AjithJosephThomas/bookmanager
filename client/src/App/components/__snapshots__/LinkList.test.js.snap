// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkList component renders LinkList component to render with empty list 1`] = `
<ul
  className="sc-AxirZ cjNxnn"
/>
`;

exports[`LinkList component renders LinkList component to render with list 1`] = `
<ul
  className="sc-AxirZ cjNxnn"
>
  <li>
    <a
      href="/1"
      onClick={[Function]}
    >
      test
    </a>
  </li>
</ul>
`;
